enhancements/microshift/microshift-router-configuration.md

```diff
@@ -0,0 +1,480 @@
+   LoadBalancer service type for the router.
+
+### API Extensions
+As described in the proposal, there is an entire new section in the configuration:
+```yaml
+ingress:
+  status: <Enabled|Disabled> # Defaults to Enabled.
+  ports:
+    http: <int> # Defaults to 80.
+    https: <int> # Defaults to 443.
+  expose: # Defaults to IPs in the host. Details below.
+    hostnames:
+    - <hostname>
+    interfaces:
+    - <NIC name>
+    ipAddresses:
+    - <IP address>
+```
+
+For more information check each individual section.
+
+### Topology Considerations
+#### Hypershift / Hosted Control Planes
+N/A
+
+#### Standalone Clusters
+N/A
+
+#### Single-node Deployments or MicroShift
+Enhancement is solely intended for MicroShift.
+
+### Implementation Details/Notes/Constraints
+The default router is composed of a bunch of assets part of the MicroShift
+binary. These assets come from the rebase, copied from the original router in
+[cluster-ingress-operator](https://github.com/openshift/cluster-ingress-operator).
+There is already a [LoadBalancer service](https://github.com/openshift/cluster-ingress-operator/blob/master/pkg/manifests/assets/router/service-cloud.yaml)
+included, which MicroShift does not copy yet. Depending on the options
+configured in MicroShift, this resource will need further customization done
+by MicroShift's start process.
+
+The rest of the changes imply only additions. The LoadBalancer controller needs
+an expansion on its capabilities to include more IPs in the status and log any
+changes to services. All the logic to control NetworkPolicy resources is also
+new.
+
+#### How config options change manifests
+Each of the configuration options described above has a direct effect on the
+manifests that MicroShift will apply after starting.
+
+The `ingress.status` option will drive whether the router `Deployment` and the
+`Service` get created.
+```yaml
+ingress:
+  status: <Enabled|Disabled> # Defaults to Enabled.
+```
+
+The `ingress.policy.ports.http` and `ingress.policy.ports.https` will determine,
+within the `Deployment` and the `Service`, which ports get configured to be
+exposed.
+```yaml
+ingress:
+  ports:
+    http: <int> # Defaults to 80.
+    https: <int> # Defaults to 443.
+```
+
+The `ingress.policy.expose` option contains lists of hostnames, NIC names and
+IP addresses. MicroShift will translate them to IP addresses and then update
+the `status.loadBalancer` field in the `Service`. Ovnk will pick up this field
+to configure the iptables rules.
+```yaml
+ingress:
+  expose: # Defaults to IPs in the host. Details below.
+    hostnames:
+    - <hostname>
+    interfaces:
+    - <NIC name>
+    ipAddresses:
+    - <IP address>
+```
+
+### Risks and Mitigations
+Some of the features in this enhancement proposal rely on the
+[LoadBalancer controller](loadbalancer-service-support.md).
+Some of the features rely on NetworkPolicy resources, so the CNI needs
+to support that.
```

Comment:
Why relying on the new service controller and `NetworkPolicy` is a risk?

Reply:
The service controller already exists, and it is tied to ovnk as it relies on automatically picking up the IPs from the service status field to configure the iptables rules. Without this the LoadBalancer service does not work, and with that the NetworkPolicies.

Reply:
After a Slack discussion: 1) the ingress IP provisioning is custom and implemented by the referenced controller, 2) `NetworkPolicy` is not implemented in any CNI which tightly couples the implementation to the OVN-K.